nova_plugin/server.py

```diff
@@ -253,7 +253,12 @@ def rename(name):
     ctx.logger.debug(
         "server.create() server before transformations: {0}".format(server))
 
-    _handle_image_or_flavor(server, nova_client, 'image')
+    if 'block_device_mapping' in server:
```

Comment:
nova also supports "block_device_mapping_v2" (see this: https://wiki.openstack.org/wiki/BlockDeviceConfig) for backwards compatibility..